SYMPTOM_DATA_STRUCTURES.md

```diff
@@ -0,0 +1,552 @@
+
+    // Human-readable label text (can be changed)
+    // Examples: "Infrastructure failure: omit job from CR", "Cluster DNS resolution failure(s)"
+    LabelText string `gorm:"type:varchar(200);not null;uniqueIndex" json:"label_text"`
+
+    // Markdown explanation of what this label indicates
+    Description string `gorm:"type:text" json:"description"`
+
+    // Where this label should be displayed
+    // Values: "spyglass", "metrics", "component-readiness", "jaq", etc.
+    DisplayContexts pq.StringArray `gorm:"type:text[]" json:"display_contexts"`
+
+    // Metadata
+    CreatedAt time.Time `json:"created_at"`
+    UpdatedAt time.Time `json:"updated_at"`
+    DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
+}
+
+func (Label) TableName() string {
+    return "job_run_labels"
+}
+```
+
+### 2. Symptom - Symptom Definition
+
+Defines how to detect symptoms in job artifacts.
+
+```go
+// Symptom defines rules for detecting symptoms in job artifacts
+type Symptom struct {
+    gorm.Model
+
+    // Immutable identifier for this symptom
+    // Must be valid identifier (word chars, not starting with digit)
+    ID string `gorm:"primaryKey;type:varchar(100)" json:"id"`
+
+    // Human-readable summary (can be changed)
+    Summary string `gorm:"type:varchar(200);not null;uniqueIndex" json:"summary"`
+
+    // Type of matcher
+    // Simple types: "string", "regex", "jq", "xpath", "none"
+    // Compound type: "cel" (Common Expression Language against label names)
+    MatcherType string `gorm:"type:varchar(50);not null" json:"matcher_type"`
+
+    // File pattern for simple matchers (glob pattern)
+    // Examples: "**/build-log.txt", "**/e2e-timelines/**/*.json"
+    // Null for CEL matcher type
+    FilePattern string `gorm:"type:varchar(500)" json:"file_pattern,omitempty"`
+
+    // Match string - interpretation depends on MatcherType:
+    // - "string": substring to find in file
+    // - "regex": regular expression pattern
+    // - "none": ignored (just checks file existence)
+    // - "cel": CEL expression referencing applied labels (e.g. "DNSTimeout && !OperatorError")
+    MatchString string `gorm:"type:text" json:"match_string,omitempty"`
+
+    // Labels to apply when this symptom matches (typically none or one, but can be multiple)
+    LabelIDs pq.StringArray `gorm:"type:text[]" json:"label_ids"`
+
+    // Applicability filters
+    FilterReleases pq.StringArray `gorm:"type:text[]" json:"filter_releases,omitempty"` // e.g., ["4.17", "4.18"], null = all
+    FilterReleaseStatuses pq.StringArray  `gorm:"type:text[]" json:"filter_release_statuses,omitempty"` // e.g., ["Development", "Full Support"]
+    FilterProducts pq.StringArray `gorm:"type:text[]" json:"filter_products,omitempty"` // e.g., ["OCP", "OKD", "HCM"]
+
+    // Time window for applicability (null = no time restriction)
+    ValidFrom *time.Time `json:"valid_from,omitempty"`
+    ValidUntil *time.Time `json:"valid_until,omitempty"`
```

Comment:
How do you know it is only valid for a particular time range?  What if someone promises it will never happen again but then it does :grin: ?